clients/csharp-uwp/docs/sample-app-walkthrough.md

```diff
@@ -0,0 +1,106 @@
+
+In both cases, the activation eventually leads to a call to [DialogManager](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/DialogManager.cs).HandleSignalDetected, which executes the code to complete keyword verification.
+
+## Keyword Verification
+Once a voice agent application receives a signal from MVA, the next step is to verify that the keyword detection was valid. The model used in 1st stage keyword detection is low-power, but as a result, it has a high false accept rate. For the best experience, the audio from the 1st stage detection needs to be verified. In the sample application, this is done in two stages using DLS: 2nd stage detection, which uses a higher-power, more accurate keyword detector (running locally), followed by 3rd stage detection, which sends the signal to the cloud to be verified by an even more accurate and resource-intensive keyword detection model. Fortunately, most of this complexity is encompassed by the Speech SDK and only requires proper setup.
+
+The following steps describe how to complete keyword verification with DLS through the Speech SDK and how to interact with MVA during verification.
+
+### 8. Retrieve activation audio
+Create an [AudioGraph](https://docs.microsoft.com/en-us/uwp/api/windows.media.audio.audiograph) and pass it to the CreateAudioDeviceInputNodeAsync of the ConversationalAgentSession. This will load the graph's audio buffer with the audio *starting approximately 3 seconds before MVA detected a keyword*. In the sample app, [AgentAudioInputProvider](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/AudioInput/AgentAudioInputProvider.cs)'s InitializeFromAgentSessionAsync method retrieves the audio from the ConversationalAgentSession. AgentAudioInputProvider fires an event, DataAvailable, when the audio is ready as a stream of bytes. 
+
+Note: In the sample app, the first 2 seconds of the audio are [trimmed](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/blob/master/clients/csharp-uwp/UWPVoiceAssistantSample/DialogManager.cs#L308) before being sent to Speech Services. This is because the keyword verification may fail if the full 3 seconds from pre-detection are included in the audio.
```

Comment:
Suggest explaining the extra time, e.g. "this additional leading audio is included to accommodate a wide range of keyword lengths and speaker speeds and the initial trim should be tailored to each assistant."